include/odp/api/spec/traffic_mngr.h

```diff
@@ -331,6 +331,26 @@ typedef struct {
 	/** The per_level array specifies the TM system capabilities that
 	 * can vary based upon the tm_node level. */
 	odp_tm_level_capabilities_t per_level[ODP_TM_MAX_LEVELS];
+
+	/** node_priority indicates that this TM system supports
+	 * only having single priority associated to fan-in nodes at any
+	 * level. When TRUE, Priority is fixed for a fanin node i.e all packets
+	 * fed by a particular fanin node to a tm node will have a fixed priority.
+	 * This is equivalent to the packet priority being overloaded by
+	 * fan-in node priority. When FALSE, every fan-in node can feed packets
+	 * to parent tm node with priorities ranging from
+	 * 0..ODP_TM_MAX_PRIORITIES - 1.
+	 */
+	odp_bool_t node_priority_override;
+
+	/** max_wfq_groups_per_node indicates maximum number of wfq groups supported
+	 * by a tm node at any level. This capability is valid only when
+	 * node_priority_override is TRUE. In a given topology only these many
+	 * groups of children can be formed of same node priority under a parent.
+	 * This value can range from 1..ODP_TM_MAX_PRIORITIES as there can be
+	 * only as many number of WFQ groups as MAX priority.
+	 */
+	uint8_t max_wfq_groups_per_node;
```

Comment:
This is needed only if max_wfq_groups_per_node can be smaller than the max_priority of level capabilities. Do we see a need for that (i.e. not capping max_priority capability based on how many wfq groups can be supported)?

If we need this field, then maybe there should be a corresponding field in odp_tm_requirements_t or odp_tm_level_requirements_t, like there are for most capabilities.

Why is only for the priority_override case? If this capability value is added, why shouldn't it apply equally to the old style priority handling? If this capability field turns out to be unrelated to the priority_override setting, then I think it should be introduced in a separate patch.

There is no explanation of what WFQ groups is in the API. There is in the commit message but it should be in the API spec.

Reply:
It seems I misunderstood the meaning of max_wfq_groups_per_node. As discussed, the intention of the capability was to tell that only that many WFQ groups within a TM node can have more than one input among which WFQ is done. Whether we say that the rest of the WFQ groups can have only single input or that there are no more WFQ groups and the rest of the input do not go through WFQ schedulers at all may be just a matter of how to model and describe things in the API.